### DISCUSSION TITLE
API update commands stopped working…

### DISCUSSION THREAD

**Discussion by petehhhhh** [llamafilm/tesla-http-proxy-addon#151] [Q&A]
Thanks for the add in. Worked great for a couple of days but now my update commands have stopped working. Didn't change anything.. Reads are still fine.

I ended up hosting the public key on a separate server running Apache as just could not get NGINX to work hosting the key on HA. Literally tried for hours. Would the key being on a separate server be an issue? From i what i could see that key stays static right ? Like i say it did work fine for a day or two...

**Comment by petehhhhh** [comment]
its started working again...

**Comment by llamafilm** [comment]
You’re probably hitting the API rate limits.  https://developer.tesla.com/docs/fleet-api/getting-started/subscription-plans#discovery

**Comment by petehhhhh** [comment]
ah ok. Thought they weren't enforcing that. Very limiting for updates. Eg i use to control charging of car to align with available solar.  That won't work with eg 5 requests per vehicle / day....

**Comment by iainbullock** [comment]
@petehhhhh Yes the API limit is no good if you are using it to charge according to solar. You might be interested in our project to use Bluetooth to send commands to the car. There are no rate limits. It will work alongside this project

https://github.com/tesla-local-control

**Comment by petehhhhh** [comment]
thanks Iain. That looks better... New to this LFE world. How do i run the bluetooth standalone (in garage) so can connect to it via network ? Is there a standalone module i can use ?

**Comment by iainbullock** [comment]
Yes you can use the Docker version for this purpose. https://github.com/tesla-local-control/tesla_ble_mqtt_docker

I run mine standalone on a Raspberry Pi 3b+ which is remote from my main Home Assistant setup. It has built in bluetooth to talk to the car, and Wifi to connect to my network. Note Tesla libraries are somewhat fussy about which bluetooth adapters work. The RPi3b+ has built in BLE and 2.4/5GHz Wifi which works fine

If you need support, raise an Issue on the tesla_ble_mqtt_docker repo

**Reply by petehhhhh** [reply]
Excellent thx Iain. Have ordered a Raspberry 5 to upgrade my HA and will use the old one for this. Documentation looks good so hopefully can work it out. The code in GitHub looks to be all shell script? There is an underlying app on the container or how's it working ?

**Comment by iainbullock** [comment]
Yes it is mostly written in shell, but uses Tesla libraries which are written in golang and compiled as executables as part of the build process. See https://github.com/teslamotors/vehicle-command
